libs/react-helpers/src/lib/format/number.tsx

```diff
@@ -100,3 +100,67 @@ export const useNumberParts = (
 export const isNumeric = (
   value?: string | number | BigNumber | null
 ): value is NonNullable<number | string> => /^-?\d*\.?\d+$/.test(String(value));
+
+const INFINITY = '∞';
+const DEFAULT_COMPACT_ABOVE = 1_000_000;
+const DEFAULT_COMPACT_CAP = new BigNumber(1e24);
+/**
+ * Compacts given number to human readable format.
+ * @param number
+ * @param decimals Number of decimal places
+ * @param compactDisplay Display mode; short -> 1e6 == 1M; ling -> 1e6 1 million
+ * @param compactAbove Compact number above threshold
+ * @param cap Use scientific notation above threshold
+ */
+export const compactNumber = (
+  number: BigNumber,
+  decimals: number | 'infer' = 'infer',
+  compactDisplay: 'short' | 'long' = 'short',
+  compactAbove = DEFAULT_COMPACT_ABOVE,
+  cap = DEFAULT_COMPACT_CAP
+) => {
+  if (!number.isFinite()) return `${number.isNegative() ? '-' : ''}${INFINITY}`;
+
+  const decimalPlaces =
+    (decimals === 'infer' ? number.decimalPlaces() : decimals) || 0;
+
+  if (number.isLessThan(DEFAULT_COMPACT_ABOVE)) {
+    return formatNumber(number, decimalPlaces);
+  }
+
+  /**
+   * Note: it compacts number up to 1_000_000_000_000 (1e12) -> 1T, all above is formatted as iteration of T.
+   * Example: 1.579208923731619e59 -> 157,920,892,373,161,900,000,000,000,000,000,000,000,000,000,000T
+   */
+  const compactNumFormat = new Intl.NumberFormat(getUserLocale(), {
+    minimumFractionDigits: Math.max(0, decimalPlaces),
+    maximumFractionDigits: Math.max(0, decimalPlaces),
+    notation: 'compact',
+    compactDisplay,
+  });
+  const scientificNumFormat = new Intl.NumberFormat(getUserLocale(), {
+    minimumFractionDigits: Math.max(0, decimalPlaces),
+    maximumFractionDigits: Math.max(0, decimalPlaces),
+    notation: 'scientific',
+  });
+
+  if (number.isGreaterThan(DEFAULT_COMPACT_CAP)) {
+    const r = /E(\d+)$/i;
+    const formatted = scientificNumFormat.format(Number(number));
+    const eNotation = formatted.match(r);
+    if (eNotation && eNotation.length > 1) {
+      const power = eNotation[1];
+      return (
```

Comment:
chore (non blocking): Seeing as this can return jsx I think it would be clearer if this just became a CompactNumber component. Or alternatively if the compactNumber function did not return jsx, and the rendering of html was left up to the component that uses this function.